opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/hystrix/HystrixTraceCommandTest.java

```diff
@@ -0,0 +1,182 @@
+            return "Hello!!!";
+        }
+
+        @HystrixCommand(fallbackMethod = "defaultGreeting")
+        public String sayHelloToUser(String user) {
+            Span span = mockTracer.buildSpan("sayHelloToUser")
+                    .startManual();
+            try {
+                throw new IllegalStateException("Thrown Purposefully");
+            } finally {
+                span.finish();
+            }
+        }
+
+        public String defaultGreeting(String user) {
+            Span span = mockTracer.buildSpan("sayHelloToUser")
+                    .withTag("myTag", "fallback")
+                    .startManual();
+            try {
+                return "Hi(fallback)";
+            } finally {
+                span.finish();
+            }
+        }
+    }
+
+    @Autowired
+    private MockTracer mockTracer;
+
+    @Autowired
+    private GreetingService greetingService;
+
+    @BeforeClass
+    @AfterClass
+    public static void reset() {
+        HystrixPlugins.reset();
+    }
+
+    @Before
+    public void before() {
+        mockTracer.reset();
+    }
+
+    @Test
+    public void test_without_circuit_breaker() throws Exception {
+
+        try (ActiveSpan activeSpan = mockTracer.buildSpan("test_without_circuit_breaker")
+                .startActive()) {
+            String response = greetingService.sayHello();
+            assertThat(response).isNotNull();
+            await().atMost(5, TimeUnit.SECONDS).until(() -> mockTracer.finishedSpans().size() == 1);
+
+            List<MockSpan> mockSpans = mockTracer.finishedSpans();
+            assertEquals(1, mockSpans.size());
+            TestUtils.assertSameTraceId(mockSpans);
+        }
+    }
+
+    @Test
+    public void test_with_circuit_breaker() {
+        try (ActiveSpan activeSpan = mockTracer.buildSpan("test_with_circuit_breaker")
+                .startActive()) {
+            String response = greetingService.sayHelloToUser("tomandjerry");
+            assertThat(response).isNotNull();
+            await().until(() -> mockTracer.finishedSpans().size() == 2);
+
+            List<MockSpan> mockSpans = mockTracer.finishedSpans();
+            assertEquals(2, mockSpans.size());
+            TestUtils.assertSameTraceId(mockSpans);
+
+            assertThat(mockSpans.get(1).tags()).containsValues("fallback");
+        }
+    }
+
+    @Test
+    public void test_hystrix_trace_command() throws Exception {
+        try (ActiveSpan activeSpan = mockTracer.buildSpan("test_with_circuit_breaker")
+                .startActive()) {
+            String groupKey = "test_hystrix";
+            String commandKey = "hystrix_trace_command";
+            com.netflix.hystrix.HystrixCommand.Setter setter = com.netflix.hystrix.HystrixCommand.Setter
+                    .withGroupKey(HystrixCommandGroupKey.Factory.asKey(groupKey))
+                    .andCommandKey(HystrixCommandKey.Factory.asKey(commandKey));
+            new HystrixTraceCommand<Void>(mockTracer, setter) {
```

Comment:
could we have this test but use normal `HystrixCommand`. I suspect it will be traced?

Reply:
no..  the tracing happens only with javanica HystrixCommand is what will be handled